rig-postgres/examples/vector_search_postgres.rs

```diff
@@ -0,0 +1,99 @@
+#[derive(Embed, Serialize, Deserialize, Clone, Debug, Eq, PartialEq, Default)]
+struct WordDefinition {
+    id: String,
+    word: String,
+    #[serde(skip)] // we don't want to serialize this field, we use only to create embeddings
+    #[embed]
+    definitions: Vec<String>,
+}
+
+impl std::fmt::Display for WordDefinition {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.word)
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // load environment variables from .env file
+    dotenvy::dotenv().ok();
+
+    // Create OpenAI client
+    let openai_client = rig::providers::openai::Client::from_env();
+    let model = openai_client.embedding_model(rig::providers::openai::TEXT_EMBEDDING_3_SMALL);
+
+    // setup Postgres
+    let database_url = std::env::var("DATABASE_URL").expect("DATABASE_URL not set");
+    let pool = PgPoolOptions::new()
+        .max_connections(50)
+        .idle_timeout(std::time::Duration::from_secs(5))
+        .connect(&database_url)
+        .await
+        .expect("Failed to create postgres pool");
+
+    // make sure database is setup
+    sqlx::migrate!("./examples/migrations").run(&pool).await?;
+
+    // create test documents with mocked embeddings
+    let words = vec![
+        WordDefinition {
+            id: "doc0".to_string(),
+            word: "flurbo".to_string(),
+            definitions: vec![
+                "1. *flurbo* (name): A flurbo is a green alien that lives on cold planets.".to_string(),
+                "2. *flurbo* (name): A fictional digital currency that originated in the animated series Rick and Morty.".to_string()
+            ]
+        },
+        WordDefinition {
+            id: "doc1".to_string(),
+            word: "glarb-glarb".to_string(),
+            definitions: vec![
+                "1. *glarb-glarb* (noun): A glarb-glarb is a ancient tool used by the ancestors of the inhabitants of planet Jiro to farm the land.".to_string(),
+                "2. *glarb-glarb* (noun): A fictional creature found in the distant, swampy marshlands of the planet Glibbo in the Andromeda galaxy.".to_string()
+            ]
+        },
+        WordDefinition {
+            id: "doc2".to_string(),
+            word: "linglingdong".to_string(),
+            definitions: vec![
+                "1. *linglingdong* (noun): A term used by inhabitants of the far side of the moon to describe humans.".to_string(),
+                "2. *linglingdong* (noun): A rare, mystical instrument crafted by the ancient monks of the Nebulon Mountain Ranges on the planet Quarm.".to_string()
+            ]
+        }];
+
+    let documents = EmbeddingsBuilder::new(model.clone())
+        .documents(words)
+        .unwrap()
+        .build()
+        .await
+        .expect("Failed to create embeddings");
+
+    // init vector store
+    sqlx::query("TRUNCATE documents").execute(&pool).await?;
```

Comment:
Perhaps indicate that this query deletes existing documents in the DB, wouldn't want someone to run this thinking it's required

Reply:
Added a better comment, good point